### PR TITLE
add kfp v1 importer docs

### PR DESCRIPTION
**Pull request by connor-mccarthy**
Adds `v2.dsl.importer` docs for KFP v1. Includes discussion of dynamic importer metadata.

### CODE REVIEW

File: content/en/docs/components/pipelines/sdk-v2/importer-component.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
++++
+title = "Importer component"
```
Comment (chensun) [inline content/en/docs/components/pipelines/sdk-v2/importer-component.md:2]: Maybe hold off on this doc for now. With the upcoming pipeline input artifact support, I personally had a hard time to justify if importer would still be needed. Maybe there're cases I haven't thought about where importer is still useful. Happy to chat more offline.

Comment (chensun) [reply content/en/docs/components/pipelines/sdk-v2/importer-component.md:2]: Synced offline, we can merge this when ready given importer is already exposed to existing users, this just completes the documentation part. We can discuss the usage of importer later.

**Comment by connor-mccarthy** [comment]
/hold
Merge after https://github.com/kubeflow/pipelines/pull/8226 merges

**Review by chensun — COMMENTED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**